src/test/java/io/github/thebusybiscuit/slimefun4/utils/TestNumberUtils.java

```diff
@@ -91,4 +91,29 @@ void testCompactDecimals() {
         Assertions.assertEquals("-2Q", NumberUtils.getCompactDouble(-2000000000000000.0));
     }
 
+    @Test
+    @DisplayName("Test flow safe addition")
+    void testFlowSafeAddition() {
+        Assertions.assertEquals(Integer.MAX_VALUE, NumberUtils.flowSafeAddition(Integer.MAX_VALUE, 1));
+        Assertions.assertEquals(Integer.MAX_VALUE, NumberUtils.flowSafeAddition(1, Integer.MAX_VALUE));
+        Assertions.assertEquals(Integer.MAX_VALUE, NumberUtils.flowSafeAddition(Integer.MAX_VALUE - 1, 2));
+        Assertions.assertEquals(Integer.MAX_VALUE, NumberUtils.flowSafeAddition(2, Integer.MAX_VALUE - 1));
+        Assertions.assertEquals(Integer.MIN_VALUE, NumberUtils.flowSafeAddition(Integer.MIN_VALUE, -1));
+        Assertions.assertEquals(Integer.MIN_VALUE, NumberUtils.flowSafeAddition(-1, Integer.MIN_VALUE));
+        Assertions.assertEquals(Integer.MIN_VALUE, NumberUtils.flowSafeAddition(Integer.MIN_VALUE + 1, -2));
+        Assertions.assertEquals(Integer.MIN_VALUE, NumberUtils.flowSafeAddition(-2, Integer.MIN_VALUE + 1));
+    }
+    
+    @Test
+    @DisplayName("Test limited addition")
+    void testLimitedAddition() {
+        Assertions.assertEquals(1000, NumberUtils.limitedAddition(1000, 1, 1000, -1000));
+        Assertions.assertEquals(1000, NumberUtils.limitedAddition(1, 1000, 1000, -1000));
+        Assertions.assertEquals(1000, NumberUtils.limitedAddition(999, 2, 1000, -1000));
+        Assertions.assertEquals(1000, NumberUtils.limitedAddition(2, 999, 1000, -1000));
+        Assertions.assertEquals(-1000, NumberUtils.limitedAddition(-1000, -1, 1000, -1000));
+        Assertions.assertEquals(-1000, NumberUtils.limitedAddition(-1, -1000, 1000, -1000));
+        Assertions.assertEquals(-1000, NumberUtils.limitedAddition(-999, -2, 1000, -1000));
+        Assertions.assertEquals(-1000, NumberUtils.limitedAddition(-2, -999, 1000, -1000));
+    }
```

Comment:
This basically cries for a `CsvSource` input. But then again, we always did it this way thus far so why change now XD

Does this cover all possible cases for full coverage?

Reply:
It should cover all cases, I just went through the 2 booleans and did the calls that would set whichever parts of it to be true

Reply:
Also ngl don't know what a csvSource is

Reply:
Me neither.

*Just kidding*, it's one way to insert lots of different parameters into parameterized tests by formatting the data like a .csv file. But it's not needed here at all, we never go that fancy.